### ISSUE TITLE
HMSG parsing fails when message body is empty

### ISSUE THREAD

**Issue by rutgerbrf** [nats-io/nats.rs#167]
Make sure that these boxes are checked before submitting your issue -- thank you!

 - [x] Included below version and environment information
 - [x] Included a [Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve)

#### NATS version (`grep 'name = "nats"' Cargo.lock -A 1`)

nats.rs commit 357cfb62
NATS Server version 2.2.1
NATS CLI version 0.0.22

#### rustc version (`rustc --version` - we support Rust 1.41 and up)

rustc 1.51.0 (2fd73fabe 2021-03-23)

#### OS/Container environment:

Manjaro 21.0.1 Ornara

#### Steps or code to reproduce the issue:

1. Start the NATS server (localhost)
2. Clone the nats.go library (b8530c78)
3. Modify the nats-sub example in the nats.go library to print headers:
    ```diff
	diff --git a/examples/nats-sub/main.go b/examples/nats-sub/main.go
	index fc3b91e..cd3f906 100644
	--- a/examples/nats-sub/main.go
	+++ b/examples/nats-sub/main.go
	@@ -14,6 +14,7 @@
	 package main
	 
	 import (
	+	"bytes"
	 	"flag"
	 	"log"
	 	"os"
	@@ -38,7 +39,9 @@ func showUsageAndExit(exitcode int) {
	 }
	 
	 func printMsg(m *nats.Msg, i int) {
	-	log.Printf("[#%d] Received on [%s]: '%s'", i, m.Subject, string(m.Data))
	+	var hbuf bytes.Buffer
	+	m.Header.Write(&hbuf)
	+	log.Printf("[#%d] Received on [%s]: '%s', headers: %s", i, m.Subject, string(m.Data), hbuf.String())
	 }
	 
	 func main() {

    ```
4. Run the nats.go nats-sub example and subscribe to a topic that you'll use (I'm using `test`):
    ```
    $ go run main.go -s localhost test
    Listening on [test]
    ```
5. Run the nats.rs nats-box example and subscribe to the same topic:
    ```
    $ cargo run --example nats-box -- --server localhost sub test
        Finished dev [unoptimized + debuginfo] target(s) in 0.08s
         Running `target/debug/examples/nats-box --server localhost sub test`
    Listening on 'test'
    ```
6. Run the following command:
    ```
    $ nats pub test "test"
    ```
    The nats-sub example from nats.go says:
    ```
    [#1] Received on [test]: 'test', headers: 
    ```
    The nats-box example from nats.rs says:
    ```
    Received a Message { subject: "test", headers: None, reply: None, length: 4 }
    ```
    Of course, that's expected behavior.
7. Run the following command:
    ```
    $ nats pub test ""
    ```
    The nats-sub example from nats.go says:
    ```
    [#2] Received on [test]: '', headers: 
    ```
    The nats-box example from nats.rs says:
    ```
    Received a Message { subject: "test", headers: None, reply: None, length: 0 }
    ```
    That's also expected behavior.
8. Run the following command:
    ```
    $ nats pub -H "Test: works" test "test"
    ```
    The nats-sub example from nats.go says:
    ```
    [#3] Received on [test]: 'test', headers: Test: works
    ```
    The nats-box example from nats.rs says:
    ```
    Received a Message { subject: "test", headers: Some(Headers { inner: {"Test": {"works"}} }), reply: None, length: 4 }
    ```
    That's also expected behavior.
9. Now run the following command (reproduce the issue):
    ```
    $ nats pub -H "Test: works" test ""
    ```

#### Expected result:

The nats-sub example from nats.go says:
```
[#4] Received on [test]: '', headers: Test: works
```

The expected result is that a message with headers and an empty body is parsed by the nats.rs library.
Or, more specifically, this output was expected:
```
Received a Message { subject: "test", headers: Some(Headers { inner: {"Test": {"works"}} }), reply: None, length: 0 }
```

#### Actual result:

The nats-box example from nats.rs fails to parse the message and does not log having received the message.

**Comment by spacejam** [comment]
Thanks for reporting this, and for your detailed issue! I'm about to publish a fix in 0.9.12 now. https://github.com/nats-io/nats.rs/pull/170

**Comment by spacejam** [comment]
0.9.12 has been published which addresses this issue. Thanks again for the thorough report, which made fixing it quite easy :)
